packages/puppeteer-core/src/common/bidi/Connection.ts

```diff
@@ -160,15 +160,24 @@ export class Connection extends EventEmitter {
     method: T,
     params: Commands[T]['params']
   ): Promise<{result: Commands[T]['returnType']}> {
-    return this.#callbacks.create(method, this.#timeout, id => {
-      const stringifiedMessage = JSON.stringify({
-        id,
+    return this.#callbacks.create(
+      method,
+      this.#timeout,
+      id => {
+        const stringifiedMessage = JSON.stringify({
+          id,
+          method,
+          params,
+        } as Bidi.Command);
+        debugProtocolSend(stringifiedMessage);
+        this.#transport.send(stringifiedMessage);
+      },
+      {
         method,
         params,
-      } as Bidi.Command);
-      debugProtocolSend(stringifiedMessage);
-      this.#transport.send(stringifiedMessage);
-    }) as Promise<{result: Commands[T]['returnType']}>;
+        stack: new Error().stack,
```

Comment:
there is already an error in `Callback.#error`. Do we need to create a stack here too?

Reply:
Creating the stack here makes more sense since the callback is triggered later.

Reply:
I see that this happens at the same time as the construction of the callback (the error is also created in the constructor)